transport/controller.cc

```diff
@@ -214,13 +215,15 @@ future<> controller::do_stop_server() {
 
     try {
         co_await unsubscribe_server(server);
-        co_await server.stop();
+        co_await server.invoke_on_all([] (auto& s) { return s.shutdown(); });
     } catch (...) {
         if (!ex) {
             ex = std::current_exception();
         }
     }
 
+    (void)server.stop().finally([s = std::move(cserver), h = _bg_stops.hold()] {});
```

Comment:
I think there's a risk here, that if somebody stops the server and then immediately destroys the controller object (e.g., consider that somebody shuts down Scylla immediately after disabling the binary API), we'll have a use-after-free, no?
It looks to me like we may need to hold this future somewhere, and wait for it when closing the controller.

Reply:
No, the server itself is a unique_ptr that's std::move()-ed onto the server.stop()'s finally continuation from controller. So even if controller is gone after it, the pointer on server is already nullptr